join-domain/elx/sssd/files/join.sh

```diff
@@ -0,0 +1,111 @@
+#!/bin/bash
+set -eu
+#
+# Script to join host to domain
+#
+#################################################################
+JOIN_DOMAIN="${JOIN_DOMAIN:-UNDEF}"
+JOIN_OU="${JOIN_OU:-}"
+JOIN_USER="${JOIN_USER:-Administrator}"
+JOIN_CNAME="${JOIN_CNAME:-UNDEF}"
+PWCRYPT="${ENCRYPT_PASS:-UNDEF}"
+PWUNLOCK="${ENCRYPT_KEY:-UNDEF}"
+CLIENT_OSNAME="$(
+  awk -F "=" '/^NAME/{ print $2}' /etc/os-release |
+  sed 's/"//g'
+)"
+CLIENT_OSVERS="$(
+  awk -F "=" '/^VERSION_ID/{ print $2 }' /etc/os-release |
+  sed 's/"//g'
+)"
+
+# Get clear-text password from crypt
+function PWdecrypt {
+  local PWCLEAR
+
+  # Get cleartext password-string
+  PWCLEAR=$(
+    echo "${PWCRYPT}" | \
+    openssl enc -aes-256-cbc -md sha256 -a -d -salt -pass pass:"${PWUNLOCK}"
+  )
+
+  # This sucks, but...
+  # shellcheck disable=SC2181
+  if [[ $? -ne 0 ]]
+  then
+    echo ""
+    return 1
+  else
+    echo "${PWCLEAR}"
+    return 0
+  fi
```

Comment:
The [shellcheck docs](https://www.shellcheck.net/wiki/SC2181) suggest doing assignment inline in the if test. Maybe something like this?

```suggestion
  # Get cleartext password-string
  if PWCLEAR=$(
    echo "${PWCRYPT}" | \
    openssl enc -aes-256-cbc -md sha256 -a -d -salt -pass pass:"${PWUNLOCK}"
  )
  then
    echo "${PWCLEAR}"
    return 0
  else
    echo ""
    return 1
  fi
```

Reply:
Unfortunately, if the `openssl enc …` operation fails (e.g., because the key and encrypted-string are mismatched), `PWCLEAR` will still have _a_ value (just the _wrong_ value – a binary-stream). Had to resort to the `if [[ $? -ne 0 ]]` kludge to be extra sure that the test-statement evaluated based _solely_ on the process's return code and not simply whether `PWCLEAR` had _a_ value

Reply:
Take another look at the suggested change... My understanding is that even if openssl outputs junk in that case, it is _still_ returning an error exit, which means the if test will fail and the `else` clause will be effective and the function will echo the empty string and return 1.